varya/classes/class-varya-custom-colors.php

```diff
@@ -262,6 +267,34 @@ function varya_editor_custom_color_variables() {
 		}
 	}
 
+	/**
+	 * Customizer contrast validation warnings.
+	 * 
+	 * @author Per Soderlind
+	 * @link https://github.com/soderlind/2016-customizer-demo
+	 */
+	function on_customize_controls_enqueue_scripts() {
+		$handle = 'wcag-validate-customizer-color-contrast';
+		$src    = get_template_directory_uri() . '/assets/js/customizer-validate-wcag-color-contrast.js';
+		$deps 	= [ 'customize-controls' ];
+
+		$exports = [
+			'validate_color_contrast' => [
+				// key = current color control , values = array with color controls to check color contrast against
+				'varya_--global--color-primary' => [ "varya_--global--color-background", "varya_--global--color-background-light" ],
+				'varya_--global--color-foreground' => [ "varya_--global--color-background", "varya_--global--color-background-light" ],
+				'varya_--global--color-background' => [ "varya_--global--color-primary", "varya_--global--color-foreground" ],
+				'varya_--global--color-background-light' => [ "varya_--global--color-primary", "varya_--global--color-foreground" ],
+			],
```

Comment:
Looking at this again, I think we can simply compare all of the colors against the background-color (accept the background itself which can be compared with the foreground). Otherwise, when we compare the colors with more than just one color, it gets tricky to figure out which colors is actually causing the warning. If we change it to something like this, less warning pop up and when they do, it clearer which color option is causing it.

```suggestion
			'validate_color_contrast' => [
				// key = current color control , values = array with color controls to check color contrast against
				'varya_--global--color-primary'          => [ "varya_--global--color-background" ],
				'varya_--global--color-secondary'        => [ "varya_--global--color-background" ],
				'varya_--global--color-background-light' => [ "varya_--global--color-background" ],
				'varya_--global--color-foreground'       => [ "varya_--global--color-background" ],
				'varya_--global--color-background'       => [ "varya_--global--color-foreground" ],
			],
```

Also, we’re considering turning `varya_--global--color-background-light` into `varya_--global--color-accent` or `varya_--global--color-tertiary`. It’ll basically be a dormant utility class that child themes can tap into to add a third color if a design calls for it. The `background-light` and `background-dark` colors actually don’t make sense in the system.

Reply:
Agreed, let's only compare to the background. 👍 

Until we sort out a definite name/direction for the `background-light` option, I think we should just leave contrast tests off of that one. Otherwise it just gets too complicated. We can add that in later.